lib/tasks/say.js

```diff
@@ -1,96 +1,26 @@
-  let startPos = 0;
-  let charPos = isSSML ? 7 : 0;  // skip <speak>
-  let tag;
-  //logger.debug({isSSML}, `breakLengthyTextIfNeeded: handling text of length ${text.length}`);
-  while (startPos + charPos < text.length) {
-    if (isSSML && !tag && text[startPos + charPos] === '<') {
-      const tagStartPos = ++charPos;
-      while (startPos + charPos < text.length) {
-        if (text[startPos + charPos] === '>') {
-          if (text[startPos + charPos - 1] === '\\') tag = null;
-          else if (!tag) tag = text.substring(startPos + tagStartPos, startPos + charPos - 1);
-          break;
-        }
-        if (!tag) {
-          const c = text[startPos + charPos];
-          if (c === ' ') {
-            tag = text.substring(startPos + tagStartPos, startPos + charPos);
-            //logger.debug(`breakLengthyTextIfNeeded: enter tag ${tag} (space)`);
-            break;
-          }
-        }
-        charPos++;
-      }
-      if (tag) {
-        //search for end of tag
-        //logger.debug(`breakLengthyTextIfNeeded: searching forward for </${tag}>`);
-        const e1 = text.indexOf(`</${tag}>`, startPos + charPos);
-        const e2 = text.indexOf('/>', startPos + charPos);
-        const tagEndPos = e1 === -1 ? e2 : e2 === -1 ? e1 : Math.min(e1, e2);
-        if (tagEndPos === -1) {
-          //logger.debug(`breakLengthyTextIfNeeded: exit tag ${tag} not found, exiting`);
-        } else {
-          //logger.debug(`breakLengthyTextIfNeeded: exit tag ${tag} found at ${tagEndPos}`);
-          charPos = tagEndPos + 1;
-        }
-        tag = null;
-      }
-      continue;
-    }
-
-    if (charPos < chunkSize) {
-      charPos++;
-      continue;
-    }
-
-    // start looking for a good break point
-    let chunkIt = false;
-    const a = text[startPos + charPos];
-    const b = text[startPos + charPos + 1];
-    if (/[\.!\?]/.test(a) && /\s/.test(b)) {
-      //logger.debug('breakLengthyTextIfNeeded: breaking at sentence end');
-      chunkIt = true;
-    }
-    if (chunkIt) {
-      charPos++;
-      const chunk = text.substr(startPos, charPos);
-      if (isSSML) {
-        result.push(0 === startPos ? `${chunk}</speak>` : `<speak>${chunk}</speak>`);
-      }
-      else result.push(chunk);
-      charPos = 0;
-      startPos += chunk.length;
-
-      //logger.debug({chunk: result[result.length - 1]},
-      //  `breakLengthyTextIfNeeded: chunked; new starting pos ${startPos}`);
-
-    }
-    else charPos++;
+  if (text.length <= chunkSize || !isSSML) return [text];
+  const options = {
+    // MIN length
+    softLimit: 100,
+    // MAX length, exclude 15 characters <speak></speak>
+    hardLimit: chunkSize - 15,
+    // Set of extra split characters (Optional property)
+    extraSplitChars: ',;!?',
+  };
+  pollySSMLSplit.configure(options);
+  try {
+    return pollySSMLSplit.split(text);
+  } catch (err) {
+    throw new Error(`There is error ${err} while splitting SSML, please check syntax. 
```

Comment:
I need more details on in what circumstances we throw this error.  I'm not sure we should ever throw an error, we should handle all valid ssml regardless of length.